src/prpy/planning/base.py

```diff
@@ -466,6 +466,96 @@ def plan(self, method, args, kw_args):
         else:
             raise UnsupportedPlanningError()
 
+
+class HardwareJointLimitValidator(MetaPlanner):
+    def __init__(self, planner):
+        super(HardwareJointLimitValidator, self).__init__()
+        self._planner = planner
+
+    def __str__(self):
+        return 'HardLimitValidator({:s})'.format(', '.join(map(str, self._planners)))
+
+    def get_planners(self, method_name):
+        if planner.has_planning_method(method_name):
+            return [self._planner]
+        else:
+            return []
+
+    def plan(self, method, args, kw_args):
+
+        # TODO : Is this correct?
+        robot = args[0]
+
+        #Get HARD DOF Limits from robot
+        #hard_limit_min, hard_limit_max = robot.GetHardwareDOFLimits()
```

Comment:
One idea would be to just pass the hard limits into the `HardwareJointLimitValidator` instance on construction. Then you don't need to add any additional robot APIs.